teamcity: failed tests on master: testrace/TestBackfillErrors

The following tests appear to have failed:

[#308090](https://teamcity.cockroachdb.com/viewLog.html?buildId=308090):

```
--- FAIL: testrace/TestBackfillErrors (0.000s)
Race detected!

------- Stdout: -------
W170728 16:13:00.265855 100412 server/server.go:303  [n?] all stores are configured as in-memory stores, so not setting up a temporary store. Queries with working set larger than memory will fail
W170728 16:13:00.267138 100412 server/status/runtime.go:111  Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
I170728 16:13:00.290668 100412 server/config.go:523  [n?] 1 storage engine initialized
I170728 16:13:00.290726 100412 server/config.go:525  [n?] RocksDB cache size: 512 MiB
I170728 16:13:00.290758 100412 server/config.go:525  [n?] store 0: in-memory, size 100 MiB
I170728 16:13:00.315864 100412 server/node.go:336  [n?] **** cluster 64632d39-1553-4759-a284-81cce5e7124b has been created
I170728 16:13:00.315989 100412 server/server.go:781  [n?] **** add additional nodes by specifying --join=127.0.0.1:33588
I170728 16:13:00.336973 100412 storage/store.go:1262  [n1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I170728 16:13:00.337539 100412 server/node.go:421  [n1] initialized store [n1,s1]: {Capacity:536870912 Available:536870912 RangeCount:1 LeaseCount:1 WritesPerSecond:100.15556663387406}
I170728 16:13:00.337789 100412 server/node.go:320  [n1] node ID 1 initialized
I170728 16:13:00.338000 100412 gossip/gossip.go:297  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:33588" > attrs:<> locality:<> 
I170728 16:13:00.338982 100412 storage/stores.go:295  [n1] read 0 node addresses from persistent storage
I170728 16:13:00.340302 100412 server/node.go:556  [n1] connecting to gossip network to verify cluster ID...
I170728 16:13:00.340497 100412 server/node.go:581  [n1] node connected via gossip and verified as part of cluster "64632d39-1553-4759-a284-81cce5e7124b"
I170728 16:13:00.340813 100412 server/node.go:362  [n1] node=1: started with [=] engine(s) and attributes []
I170728 16:13:00.342956 100412 sql/executor.go:419  [n1] creating distSQLPlanner with address {tcp 127.0.0.1:33588}
I170728 16:13:00.367137 100412 server/server.go:860  [n1] starting https server at 127.0.0.1:46980
I170728 16:13:00.367300 100412 server/server.go:861  [n1] starting grpc/postgres server at 127.0.0.1:33588
I170728 16:13:00.367343 100412 server/server.go:862  [n1] advertising CockroachDB node at 127.0.0.1:33588
I170728 16:13:00.453819 100412 sql/event_log.go:101  [n1] Event: "alter_table", target: 12, info: {TableName:eventlog Statement:ALTER TABLE system.eventlog ALTER COLUMN "uniqueID" SET DEFAULT uuid_v4() User:node MutationID:0 CascadeDroppedViews:[]}
I170728 16:13:00.484472 100412 sql/lease.go:360  [n1] publish: descID=12 (eventlog) version=2 mtime=2017-07-28 16:13:00.48316665 +0000 UTC
I170728 16:13:00.629834 100412 server/server.go:994  [n1] done ensuring all necessary migrations have run
I170728 16:13:00.629999 100412 server/server.go:996  [n1] serving sql connections
I170728 16:13:00.649779 100820 sql/event_log.go:101  [n1] Event: "node_join", target: 1, info: {Descriptor:{NodeID:1 Address:{NetworkField:tcp AddressField:127.0.0.1:33588} Attrs: Locality:} ClusterID:64632d39-1553-4759-a284-81cce5e7124b StartedAt:1501258380340544785 LastUp:1501258380340544785}
W170728 16:13:00.704022 100412 server/server.go:303  [n?] all stores are configured as in-memory stores, so not setting up a temporary store. Queries with working set larger than memory will fail
W170728 16:13:00.705523 100412 server/status/runtime.go:111  Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
W170728 16:13:00.710463 100412 gossip/gossip.go:1196  [n?] no incoming or outgoing connections
I170728 16:13:00.751966 100412 server/config.go:523  [n?] 1 storage engine initialized
I170728 16:13:00.752083 100412 server/config.go:525  [n?] RocksDB cache size: 512 MiB
I170728 16:13:00.752149 100412 server/config.go:525  [n?] store 0: in-memory, size 100 MiB
I170728 16:13:00.752313 100412 server/server.go:783  [n?] no stores bootstrapped and --join flag specified, awaiting init command.
I170728 16:13:00.789461 100870 gossip/client.go:131  [n?] started gossip client to 127.0.0.1:33588
I170728 16:13:00.797804 100844 gossip/server.go:234  [n1] received initial cluster-verification connection from {tcp 127.0.0.1:46628}
I170728 16:13:00.805059 100412 server/node.go:408  [n?] store [n0,s0] not bootstrapped
I170728 16:13:00.805189 100412 storage/stores.go:295  [n?] read 0 node addresses from persistent storage
I170728 16:13:00.806331 100412 storage/stores.go:314  [n?] wrote 1 node addresses to persistent storage
I170728 16:13:00.806471 100412 server/node.go:556  [n?] connecting to gossip network to verify cluster ID...
I170728 16:13:00.806592 100412 server/node.go:581  [n?] node connected via gossip and verified as part of cluster "64632d39-1553-4759-a284-81cce5e7124b"
I170728 16:13:00.819643 100412 kv/dist_sender.go:364  [n?] unable to determine this node's attributes for replica selection; node is most likely bootstrapping
I170728 16:13:00.835108 100412 server/node.go:313  [n?] new node allocated ID 2
I170728 16:13:00.835427 100412 gossip/gossip.go:297  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:46628" > attrs:<> locality:<> 
I170728 16:13:00.836168 100412 server/node.go:362  [n2] node=2: started with [=] engine(s) and attributes []
I170728 16:13:00.838355 100412 sql/executor.go:419  [n2] creating distSQLPlanner with address {tcp 127.0.0.1:46628}
I170728 16:13:00.855094 100778 storage/stores.go:314  [n1] wrote 1 node addresses to persistent storage
I170728 16:13:00.869082 100891 server/node.go:537  [n2] bootstrapped store [n2,s2]
I170728 16:13:00.903385 100412 server/server.go:860  [n2] starting https server at 127.0.0.1:49779
I170728 16:13:00.903979 100412 server/server.go:861  [n2] starting grpc/postgres server at 127.0.0.1:46628
I170728 16:13:00.904034 100412 server/server.go:862  [n2] advertising CockroachDB node at 127.0.0.1:46628
I170728 16:13:00.930921 100412 server/server.go:994  [n2] done ensuring all necessary migrations have run
I170728 16:13:00.931048 100412 server/server.go:996  [n2] serving sql connections
I170728 16:13:01.011876 100848 sql/event_log.go:101  [n2] Event: "node_join", target: 2, info: {Descriptor:{NodeID:2 Address:{NetworkField:tcp AddressField:127.0.0.1:46628} Attrs: Locality:} ClusterID:64632d39-1553-4759-a284-81cce5e7124b StartedAt:1501258380835911435 LastUp:1501258380835911435}
W170728 16:13:01.048874 100412 server/server.go:303  [n?] all stores are configured as in-memory stores, so not setting up a temporary store. Queries with working set larger than memory will fail
W170728 16:13:01.050162 100412 server/status/runtime.go:111  Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
W170728 16:13:01.054940 100412 gossip/gossip.go:1196  [n?] no incoming or outgoing connections
I170728 16:13:01.077858 100412 server/config.go:523  [n?] 1 storage engine initialized
I170728 16:13:01.077968 100412 server/config.go:525  [n?] RocksDB cache size: 512 MiB
I170728 16:13:01.078017 100412 server/config.go:525  [n?] store 0: in-memory, size 100 MiB
I170728 16:13:01.078199 100412 server/server.go:783  [n?] no stores bootstrapped and --join flag specified, awaiting init command.
I170728 16:13:01.109046 100939 gossip/client.go:131  [n?] started gossip client to 127.0.0.1:33588
I170728 16:13:01.113688 101095 gossip/server.go:234  [n1] received initial cluster-verification connection from {tcp 127.0.0.1:37830}
I170728 16:13:01.117908 100412 server/node.go:408  [n?] store [n0,s0] not bootstrapped
I170728 16:13:01.121219 100412 storage/stores.go:295  [n?] read 0 node addresses from persistent storage
I170728 16:13:01.121540 100412 storage/stores.go:314  [n?] wrote 2 node addresses to persistent storage
I170728 16:13:01.121642 100412 server/node.go:556  [n?] connecting to gossip network to verify cluster ID...
I170728 16:13:01.121759 100412 server/node.go:581  [n?] node connected via gossip and verified as part of cluster "64632d39-1553-4759-a284-81cce5e7124b"
I170728 16:13:01.132934 100412 kv/dist_sender.go:364  [n?] unable to determine this node's attributes for replica selection; node is most likely bootstrapping
I170728 16:13:01.140664 100412 server/node.go:313  [n?] new node allocated ID 3
I170728 16:13:01.140973 100412 gossip/gossip.go:297  [n3] NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:37830" > attrs:<> locality:<> 
I170728 16:13:01.141745 100412 server/node.go:362  [n3] node=3: started with [=] engine(s) and attributes []
I170728 16:13:01.144035 100412 sql/executor.go:419  [n3] creating distSQLPlanner with address {tcp 127.0.0.1:37830}
I170728 16:13:01.151799 101032 storage/stores.go:314  [n1] wrote 2 node addresses to persistent storage
I170728 16:13:01.155570 101082 storage/stores.go:314  [n2] wrote 2 node addresses to persistent storage
I170728 16:13:01.187958 100412 server/server.go:860  [n3] starting https server at 127.0.0.1:52285
I170728 16:13:01.188092 100412 server/server.go:861  [n3] starting grpc/postgres server at 127.0.0.1:37830
I170728 16:13:01.188139 100412 server/server.go:862  [n3] advertising CockroachDB node at 127.0.0.1:37830
I170728 16:13:01.201182 100412 server/server.go:994  [n3] done ensuring all necessary migrations have run
I170728 16:13:01.201312 100412 server/server.go:996  [n3] serving sql connections
I170728 16:13:01.212035 101127 server/node.go:537  [n3] bootstrapped store [n3,s3]
I170728 16:13:01.288295 101282 sql/event_log.go:101  [n3] Event: "node_join", target: 3, info: {Descriptor:{NodeID:3 Address:{NetworkField:tcp AddressField:127.0.0.1:37830} Attrs: Locality:} ClusterID:64632d39-1553-4759-a284-81cce5e7124b StartedAt:1501258381141490102 LastUp:1501258381141490102}
W170728 16:13:01.325969 100412 server/server.go:303  [n?] all stores are configured as in-memory stores, so not setting up a temporary store. Queries with working set larger than memory will fail
W170728 16:13:01.327200 100412 server/status/runtime.go:111  Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
W170728 16:13:01.340711 100412 gossip/gossip.go:1196  [n?] no incoming or outgoing connections
I170728 16:13:01.346154 100412 server/config.go:523  [n?] 1 storage engine initialized
I170728 16:13:01.346288 100412 server/config.go:525  [n?] RocksDB cache size: 512 MiB
I170728 16:13:01.346347 100412 server/config.go:525  [n?] store 0: in-memory, size 100 MiB
I170728 16:13:01.346556 100412 server/server.go:783  [n?] no stores bootstrapped and --join flag specified, awaiting init command.
I170728 16:13:01.425130 101263 gossip/client.go:131  [n?] started gossip client to 127.0.0.1:33588
I170728 16:13:01.426212 101348 gossip/server.go:234  [n1] received initial cluster-verification connection from {tcp 127.0.0.1:56122}
I170728 16:13:01.434684 100412 server/node.go:408  [n?] store [n0,s0] not bootstrapped
I170728 16:13:01.437116 100412 storage/stores.go:295  [n?] read 0 node addresses from persistent storage
I170728 16:13:01.437453 100412 storage/stores.go:314  [n?] wrote 3 node addresses to persistent storage
I170728 16:13:01.437549 100412 server/node.go:556  [n?] connecting to gossip network to verify cluster ID...
I170728 16:13:01.437665 100412 server/node.go:581  [n?] node connected via gossip and verified as part of cluster "64632d39-1553-4759-a284-81cce5e7124b"
I170728 16:13:01.452513 100412 kv/dist_sender.go:364  [n?] unable to determine this node's attributes for replica selection; node is most likely bootstrapping
I170728 16:13:01.457625 100412 server/node.go:313  [n?] new node allocated ID 4
I170728 16:13:01.457933 100412 gossip/gossip.go:297  [n4] NodeDescriptor set to node_id:4 address:<network_field:"tcp" address_field:"127.0.0.1:56122" > attrs:<> locality:<> 
I170728 16:13:01.458826 100412 server/node.go:362  [n4] node=4: started with [=] engine(s) and attributes []
I170728 16:13:01.461307 100412 sql/executor.go:419  [n4] creating distSQLPlanner with address {tcp 127.0.0.1:56122}
I170728 16:13:01.478446 101102 storage/stores.go:314  [n1] wrote 3 node addresses to persistent storage
I170728 16:13:01.481661 101364 storage/stores.go:314  [n3] wrote 3 node addresses to persistent storage
I170728 16:13:01.481876 101051 storage/stores.go:314  [n2] wrote 3 node addresses to persistent storage
I170728 16:13:01.521690 100412 server/server.go:860  [n4] starting https server at 127.0.0.1:46981
I170728 16:13:01.531423 100412 server/server.go:861  [n4] starting grpc/postgres server at 127.0.0.1:56122
I170728 16:13:01.531517 101322 server/node.go:537  [n4] bootstrapped store [n4,s4]
I170728 16:13:01.534556 100412 server/server.go:862  [n4] advertising CockroachDB node at 127.0.0.1:56122
I170728 16:13:01.546902 100412 server/server.go:994  [n4] done ensuring all necessary migrations have run
I170728 16:13:01.547028 100412 server/server.go:996  [n4] serving sql connections
I170728 16:13:01.609749 101356 sql/event_log.go:101  [n4] Event: "node_join", target: 4, info: {Descriptor:{NodeID:4 Address:{NetworkField:tcp AddressField:127.0.0.1:56122} Attrs: Locality:} ClusterID:64632d39-1553-4759-a284-81cce5e7124b StartedAt:1501258381458519845 LastUp:1501258381458519845}
W170728 16:13:01.699261 100412 server/server.go:303  [n?] all stores are configured as in-memory stores, so not setting up a temporary store. Queries with working set larger than memory will fail
W170728 16:13:01.701656 100412 server/status/runtime.go:111  Could not parse build timestamp: parsing time "" as "2006/01/02 15:04:05": cannot parse "" as "2006"
W170728 16:13:01.709439 100412 gossip/gossip.go:1196  [n?] no incoming or outgoing connections
I170728 16:13:01.721982 100412 server/config.go:523  [n?] 1 storage engine initialized
I170728 16:13:01.722093 100412 server/config.go:525  [n?] RocksDB cache size: 512 MiB
I170728 16:13:01.722137 100412 server/config.go:525  [n?] store 0: in-memory, size 100 MiB
I170728 16:13:01.722306 100412 server/server.go:783  [n?] no stores bootstrapped and --join flag specified, awaiting init command.
I170728 16:13:01.791963 101526 gossip/client.go:131  [n?] started gossip client to 127.0.0.1:33588
I170728 16:13:01.799253 101485 gossip/server.go:234  [n1] received initial cluster-verification connection from {tcp 127.0.0.1:35093}
I170728 16:13:01.808693 100412 server/node.go:408  [n?] store [n0,s0] not bootstrapped
I170728 16:13:01.808811 100412 storage/stores.go:295  [n?] read 0 node addresses from persistent storage
I170728 16:13:01.809113 100412 storage/stores.go:314  [n?] wrote 4 node addresses to persistent storage
I170728 16:13:01.809214 100412 server/node.go:556  [n?] connecting to gossip network to verify cluster ID...
I170728 16:13:01.809316 100412 server/node.go:581  [n?] node connected via gossip and verified as part of cluster "64632d39-1553-4759-a284-81cce5e7124b"
I170728 16:13:01.822645 100412 kv/dist_sender.go:364  [n?] unable to determine this node's attributes for replica selection; node is most likely bootstrapping
I170728 16:13:01.827629 100412 server/node.go:313  [n?] new node allocated ID 5
I170728 16:13:01.827915 100412 gossip/gossip.go:297  [n5] NodeDescriptor set to node_id:5 address:<network_field:"tcp" address_field:"127.0.0.1:35093" > attrs:<> locality:<> 
I170728 16:13:01.830031 100412 server/node.go:362  [n5] node=5: started with [=] engine(s) and attributes []
I170728 16:13:01.835329 101485 gossip/server.go:285  [n1] refusing gossip from node 5 (max 3 conns); forwarding to 3 ({tcp 127.0.0.1:37830})
I170728 16:13:01.839308 100412 sql/executor.go:419  [n5] creating distSQLPlanner with address {tcp 127.0.0.1:35093}
I170728 16:13:01.842621 101526 gossip/client.go:136  [n5] closing client to node 1 (127.0.0.1:33588): received forward from node 1 to 3 (127.0.0.1:37830)
I170728 16:13:01.844174 101501 storage/stores.go:314  [n1] wrote 4 node addresses to persistent storage
I170728 16:13:01.851673 101565 gossip/gossip.go:1210  [n5] node has connected to cluster via gossip
I170728 16:13:01.852244 101565 storage/stores.go:314  [n5] wrote 4 node addresses to persistent storage
I170728 16:13:01.854113 101502 storage/stores.go:314  [n3] wrote 4 node addresses to persistent storage
I170728 16:13:01.856214 101514 storage/stores.go:314  [n4] wrote 4 node addresses to persistent storage
I170728 16:13:01.857254 101574 storage/stores.go:314  [n2] wrote 4 node addresses to persistent storage
I170728 16:13:01.934515 100412 server/server.go:860  [n5] starting https server at 127.0.0.1:54292
I170728 16:13:01.934654 100412 server/server.go:861  [n5] starting grpc/postgres server at 127.0.0.1:35093
I170728 16:13:01.934728 100412 server/server.go:862  [n5] advertising CockroachDB node at 127.0.0.1:35093
I170728 16:13:01.936275 101600 server/node.go:537  [n5] bootstrapped store [n5,s5]
I170728 16:13:01.949563 100412 server/server.go:994  [n5] done ensuring all necessary migrations have run
I170728 16:13:01.949717 100412 server/server.go:996  [n5] serving sql connections
I170728 16:13:01.998486 101753 sql/event_log.go:101  [n5] Event: "node_join", target: 5, info: {Descriptor:{NodeID:5 Address:{NetworkField:tcp AddressField:127.0.0.1:35093} Attrs: Locality:} ClusterID:64632d39-1553-4759-a284-81cce5e7124b StartedAt:1501258381828443746 LastUp:1501258381828443746}
I170728 16:13:02.053299 101372 gossip/client.go:131  [n5] started gossip client to 127.0.0.1:37830
I170728 16:13:02.178494 101783 sql/event_log.go:101  [client=127.0.0.1:44014,user=root,n1] Event: "create_database", target: 50, info: {DatabaseName:t Statement:CREATE DATABASE t User:root}
I170728 16:13:02.218201 101783 sql/event_log.go:101  [client=127.0.0.1:44014,user=root,n1] Event: "create_table", target: 51, info: {TableName:t.test Statement:CREATE TABLE t.test (k INT PRIMARY KEY, v INT) User:root}
I170728 16:13:05.035553 101910 storage/replica_command.go:2673  [n1,s1,r1/1:/M{in-ax}] initiating a split of this range at key /Table/51/1/3200 [r2]
I170728 16:13:05.123516 101834 storage/replica_raftstorage.go:511  [n1,s1,r2/1:/{Table/51/1/3���-Max}] generated preemptive snapshot 83cd3359 at index 11
I170728 16:13:05.195696 101834 storage/store.go:3477  [n1,s1,r2/1:/{Table/51/1/3���-Max}] streamed snapshot to (n5,s5):?: kv pairs: 810, log entries: 1, rate-limit: 2.0 MiB/sec, 8ms
I170728 16:13:05.197066 101851 storage/replica_raftstorage.go:720  [n5,s5,r2/?:{-}] applying preemptive snapshot at index 11 (id=83cd3359, encoded size=22069, 1 rocksdb batches, 1 log entries)
I170728 16:13:05.198666 101851 storage/replica_raftstorage.go:728  [n5,s5,r2/?:/{Table/51/1/3���-Max}] applied preemptive snapshot in 1ms [clear=0ms batch=0ms entries=0ms commit=1ms]
I170728 16:13:05.202783 101834 storage/replica_command.go:3606  [n1,s1,r2/1:/{Table/51/1/3���-Max}] change replicas (ADD_REPLICA (n5,s5):2): read existing descriptor r2:/{Table/51/1/3200-Max} [(n1,s1):1, next=2]
I170728 16:13:05.222544 101854 storage/replica.go:2942  [n1,s1,r2/1:/{Table/51/1/3���-Max}] proposing ADD_REPLICA((n5,s5):2): updated=[(n1,s1):1 (n5,s5):2] next=3
I170728 16:13:05.236749 101957 storage/raft_transport.go:453  [n5] raft transport stream to node 1 established
I170728 16:13:05.259703 101870 storage/replica_command.go:2673  [n1,s1,r1/1:/{Min-Table/51/1/3���}] initiating a split of this range at key /Table/51/1/2400 [r3]
I170728 16:13:05.271047 101281 storage/replica_proposal.go:454  [n5,s5,r2/2:/{Table/51/1/3���-Max}] new range lease repl=(n5,s5):2 start=1501258385.243222910,0 epo=1 pro=1501258385.243231110,0 following repl=(n1,s1):1 start=0.000000000,0 exp=1501258393.875936231,0 pro=1501258384.876059132,0
I170728 16:13:05.335448 101947 storage/replica_raftstorage.go:511  [n1,s1,r3/1:/Table/51/1/{2400-3200}] generated preemptive snapshot 1ab56f66 at index 11
I170728 16:13:05.429966 101947 storage/store.go:3477  [n1,s1,r3/1:/Table/51/1/{2400-3200}] streamed snapshot to (n4,s4):?: kv pairs: 812, log entries: 1, rate-limit: 2.0 MiB/sec, 11ms
I170728 16:13:05.431908 102008 storage/replica_raftstorage.go:720  [n4,s4,r3/?:{-}] applying preemptive snapshot at index 11 (id=1ab56f66, encoded size=22187, 1 rocksdb batches, 1 log entries)
I170728 16:13:05.433612 102008 storage/replica_raftstorage.go:728  [n4,s4,r3/?:/Table/51/1/{2400-3200}] applied preemptive snapshot in 2ms [clear=0ms batch=0ms entries=0ms commit=1ms]
I170728 16:13:05.437052 101947 storage/replica_command.go:3606  [n1,s1,r3/1:/Table/51/1/{2400-3200}] change replicas (ADD_REPLICA (n4,s4):2): read existing descriptor r3:/Table/51/1/{2400-3200} [(n1,s1):1, next=2]
I170728 16:13:05.457932 102043 storage/replica.go:2942  [n1,s1,r3/1:/Table/51/1/{2400-3200}] proposing ADD_REPLICA((n4,s4):2): updated=[(n1,s1):1 (n4,s4):2] next=3
I170728 16:13:05.480674 102012 storage/raft_transport.go:453  [n4] raft transport stream to node 1 established
I170728 16:13:05.498874 101089 storage/replica_proposal.go:454  [n4,s4,r3/2:/Table/51/1/{2400-3200}] new range lease repl=(n4,s4):2 start=1501258385.490300078,0 epo=1 pro=1501258385.490307278,0 following repl=(n1,s1):1 start=0.000000000,0 exp=1501258393.875936231,0 pro=1501258384.876059132,0
I170728 16:13:05.507007 102085 storage/replica_command.go:2673  [n1,s1,r1/1:/{Min-Table/51/1/2���}] initiating a split of this range at key /Table/51/1/1600 [r4]
I170728 16:13:05.581691 102077 storage/replica_raftstorage.go:511  [n1,s1,r4/1:/Table/51/1/{1600-2400}] generated preemptive snapshot 01c1ce3f at index 11
I170728 16:13:05.659002 102077 storage/store.go:3477  [n1,s1,r4/1:/Table/51/1/{1600-2400}] streamed snapshot to (n3,s3):?: kv pairs: 810, log entries: 1, rate-limit: 2.0 MiB/sec, 8ms
I170728 16:13:05.660910 102059 storage/replica_raftstorage.go:720  [n3,s3,r4/?:{-}] applying preemptive snapshot at index 11 (id=01c1ce3f, encoded size=22135, 1 rocksdb batches, 1 log entries)
I170728 16:13:05.662552 102059 storage/replica_raftstorage.go:728  [n3,s3,r4/?:/Table/51/1/{1600-2400}] applied preemptive snapshot in 1ms [clear=0ms batch=0ms entries=0ms commit=1ms]
I170728 16:13:05.669758 102077 storage/replica_command.go:3606  [n1,s1,r4/1:/Table/51/1/{1600-2400}] change replicas (ADD_REPLICA (n3,s3):2): read existing descriptor r4:/Table/51/1/{1600-2400} [(n1,s1):1, next=2]
I170728 16:13:05.688011 102163 storage/replica.go:2942  [n1,s1,r4/1:/Table/51/1/{1600-2400}] proposing ADD_REPLICA((n3,s3):2): updated=[(n1,s1):1 (n3,s3):2] next=3
I170728 16:13:05.711131 102132 storage/raft_transport.go:453  [n3] raft transport stream to node 1 established
I170728 16:13:05.737890 101196 storage/replica_proposal.go:454  [n3,s3,r4/2:/Table/51/1/{1600-2400}] new range lease repl=(n3,s3):2 start=1501258385.726549072,0 epo=1 pro=1501258385.726557372,0 following repl=(n1,s1):1 start=0.000000000,0 exp=1501258393.875936231,0 pro=1501258384.876059132,0
I170728 16:13:05.740824 102195 storage/replica_command.go:2673  [n1,s1,r1/1:/{Min-Table/51/1/1���}] initiating a split of this range at key /Table/51/1/800 [r5]
I170728 16:13:05.811925 102108 storage/replica_raftstorage.go:511  [n1,s1,r5/1:/Table/51/1/{800-1600}] generated preemptive snapshot 7c89871d at index 11
I170728 16:13:05.900558 102108 storage/store.go:3477  [n1,s1,r5/1:/Table/51/1/{800-1600}] streamed snapshot to (n2,s2):?: kv pairs: 809, log entries: 1, rate-limit: 2.0 MiB/sec, 25ms
I170728 16:13:05.902977 102213 storage/replica_raftstorage.go:720  [n2,s2,r5/?:{-}] applying preemptive snapshot at index 11 (id=7c89871d, encoded size=22109, 1 rocksdb batches, 1 log entries)
I170728 16:13:05.905207 102213 storage/replica_raftstorage.go:728  [n2,s2,r5/?:/Table/51/1/{800-1600}] applied preemptive snapshot in 2ms [clear=0ms batch=0ms entries=0ms commit=1ms]
I170728 16:13:05.927535 102108 storage/replica_command.go:3606  [n1,s1,r5/1:/Table/51/1/{800-1600}] change replicas (ADD_REPLICA (n2,s2):2): read existing descriptor r5:/Table/51/1/{800-1600} [(n1,s1):1, next=2]
I170728 16:13:05.947583 102176 storage/replica.go:2942  [n1,s1,r5/1:/Table/51/1/{800-1600}] proposing ADD_REPLICA((n2,s2):2): updated=[(n1,s1):1 (n2,s2):2] next=3
I170728 16:13:05.973277 102263 storage/raft_transport.go:453  [n2] raft transport stream to node 1 established
I170728 16:13:06.013311 100949 storage/replica_proposal.go:454  [n2,s2,r5/2:/Table/51/1/{800-1600}] new range lease repl=(n2,s2):2 start=1501258385.998087705,0 epo=1 pro=1501258385.998094905,0 following repl=(n1,s1):1 start=0.000000000,0 exp=1501258393.875936231,0 pro=1501258384.876059132,0
I170728 16:13:06.179031 101783 sql/event_log.go:101  [client=127.0.0.1:44014,user=root,n1] Event: "create_index", target: 51, info: {TableName:test IndexName:vidx Statement:CREATE UNIQUE INDEX vidx ON t.test (v) User:root MutationID:1}
I170728 16:13:06.215335 101783 sql/lease.go:360  [client=127.0.0.1:44014,user=root,n1] publish: descID=51 (test) version=2 mtime=2017-07-28 16:13:06.211631345 +0000 UTC
I170728 16:13:06.282820 101783 sql/lease.go:286  publish (count leases): descID=51 name=test version=1 count=1
I170728 16:13:06.344698 101783 sql/lease.go:360  [client=127.0.0.1:44014,user=root,n1] publish: descID=51 (test) version=3 mtime=2017-07-28 16:13:06.343365534 +0000 UTC
I170728 16:13:06.382975 101783 sql/lease.go:286  publish (count leases): descID=51 name=test version=2 count=1
I170728 16:13:06.423135 101783 sql/lease.go:286  publish (count leases): descID=51 name=test version=2 count=1
I170728 16:13:06.503968 101783 sql/backfill.go:135  [client=127.0.0.1:44014,user=root,n1] Running backfill for "test", v=3, m=1
==================
Read at 0x00c4233bd7b8 by goroutine 940:
  runtime.slicecopy()
      /usr/local/go/src/runtime/slice.go:160 +0x0
  encoding/gob.encUint8Array()
      /usr/local/go/src/encoding/gob/encode.go:251 +0x183
  encoding/gob.(*Encoder).encodeStruct()
      /usr/local/go/src/encoding/gob/encode.go:334 +0x3e9
  encoding/gob.encOpFor.func4()
      /usr/local/go/src/encoding/gob/encode.go:587 +0xed
  encoding/gob.(*Encoder).encodeStruct()
      /usr/local/go/src/encoding/gob/encode.go:334 +0x3e9
  encoding/gob.encOpFor.func4()
      /usr/local/go/src/encoding/gob/encode.go:587 +0xed
  encoding/gob.(*Encoder).encodeStruct()
      /usr/local/go/src/encoding/gob/encode.go:334 +0x3e9
  encoding/gob.encOpFor.func4()
      /usr/local/go/src/encoding/gob/encode.go:587 +0xed
  encoding/gob.(*Encoder).encodeArray()
      /usr/local/go/src/encoding/gob/encode.go:357 +0x286
  encoding/gob.encOpFor.func1()
      /usr/local/go/src/encoding/gob/encode.go:557 +0x170
  encoding/gob.(*Encoder).encodeStruct()
      /usr/local/go/src/encoding/gob/encode.go:334 +0x3e9
  encoding/gob.(*Encoder).encode()
      /usr/local/go/src/encoding/gob/encode.go:707 +0x24f
  encoding/gob.(*Encoder).EncodeValue()
      /usr/local/go/src/encoding/gob/encoder.go:250 +0x625
  encoding/gob.(*Encoder).Encode()
      /usr/local/go/src/encoding/gob/encoder.go:175 +0x72
  github.com/cockroachdb/cockroach/pkg/kv.GRPCTransportFactory.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/transport_race.go:83 +0x438
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:195 +0x160

Previous write at 0x00c4233bd7bb by goroutine 656:
  [failed to restore the stack]

Goroutine 940 (running) created at:
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0xba
  github.com/cockroachdb/cockroach/pkg/kv.GRPCTransportFactory()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/transport_race.go:108 +0x33f
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendToReplicas()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:1135 +0x3a7
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).sendRPC()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:396 +0x352
  github.com/cockroachdb/cockroach/pkg/kv.(*DistSender).RangeLookup()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/dist_sender.go:313 +0x301
  github.com/cockroachdb/cockroach/pkg/kv.(*RangeDescriptorCache).performRangeLookup()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/range_cache.go:414 +0x4de
  github.com/cockroachdb/cockroach/pkg/kv.(*RangeDescriptorCache).lookupRangeDescriptorInternal.func3()
      /go/src/github.com/cockroachdb/cockroach/pkg/kv/range_cache.go:286 +0x11e
  github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/singleflight.(*Group).doCall()
      /go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/singleflight/singleflight.go:93 +0x4c

Goroutine 656 (finished) created at:
  github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*Flow).Start()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/flow.go:329 +0x541
  github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*flowScheduler).runFlowNow()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/flow_scheduler.go:74 +0x12c
  github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*flowScheduler).ScheduleFlow.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/flow_scheduler.go:91 +0xda
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTask()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:228 +0x104
  github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*flowScheduler).ScheduleFlow()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/flow_scheduler.go:98 +0xf3
  github.com/cockroachdb/cockroach/pkg/sql/distsqlrun.(*ServerImpl).SetupFlow()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/server.go:290 +0x2fa
  github.com/cockroachdb/cockroach/pkg/sql/distsqlrun._DistSQL_SetupFlow_Handler.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/api.pb.go:312 +0xa1
  github.com/cockroachdb/cockroach/vendor/github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc.OpenTracingServerInterceptor.func1()
      /go/src/github.com/cockroachdb/cockroach/vendor/github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc/server.go:48 +0xd6f
  github.com/cockroachdb/cockroach/pkg/sql/distsqlrun._DistSQL_SetupFlow_Handler()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/distsqlrun/api.pb.go:314 +0x1fb
  github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).processUnaryRPC()
      /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:781 +0x1221
  github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).handleStream()
      /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:981 +0x150e
  github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc.(*Server).serveStreams.func1.1()
      /go/src/github.com/cockroachdb/cockroach/vendor/google.golang.org/grpc/server.go:551 +0xb6
==================
I170728 16:08:23.353422 1 rand.go:76  Random seed: 7450721999494127540

```
Please assign, take a look and update the issue accordingly.

--- a/pkg/sql/distsqlrun/indexbackfiller.go
+++ b/pkg/sql/distsqlrun/indexbackfiller.go
@@ -215,6 +215,7 @@ func (ib *indexBackfiller) runChunk(
 		return nil, err
 	}
 
+	retried := false
 	// Write the new index values.
 	if err := ib.flowCtx.clientDB.Txn(ctx, func(ctx context.Context, txn *client.Txn) error {
 		batch := txn.NewBatch()
@@ -223,9 +224,19 @@ func (ib *indexBackfiller) runChunk(
 			// Since we're not regenerating the index entries here, if the
 			// transaction restarts the values might already have their checksums
 			// set which is invalid - clear them.
-			entry.Value.ClearChecksum()
+			if retried {
+				// Reset the value slice. This is necessary because gRPC may still be
+				// holding onto the underlying slice here. See #17348 for more details.
+				// We only need to reset RawBytes because neither entry nor entry.Value
+				// are pointer types.
+				rawBytes := entry.Value.RawBytes
+				entry.Value.RawBytes = make([]byte, len(rawBytes))
+				copy(entry.Value.RawBytes, rawBytes)
+				entry.Value.ClearChecksum()
+			}
 			batch.InitPut(entry.Key, &entry.Value, true /* failOnTombstones */)
 		}
+		retried = true
 		if err := txn.CommitInBatch(ctx, batch); err != nil {
 			if _, ok := batch.MustPErr().GetDetail().(*roachpb.ConditionFailedError); ok {
 				return pgerror.NewError(pgerror.CodeUniqueViolationError, "")

